### ISSUE TITLE
[Bug] Memory Leak While Inference Model Speech

### ISSUE THREAD

**Issue by ridhoalattas** [openvinotoolkit/openvino#12255]
##### System information (version)

- OpenVINO => 2022.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2017
- Problem classification: Model Conversion
- Framework: PyTorch (if applicable)
- Model name: Wav2vec2 (if applicable)

When i do inference with the model there is memory leak im using the newest openvino version on this case and i tried debuging all process and there is no memory leak until inference model process.

for this process im using csv file to get the data path and sentece

Here is my code to do inference 

`for idx, i in enumerate(df['path']):`
  `sound = preprocess_filewav(i)`
  `normalized_audio = model.preprocess(sound)`
  `character_probs = model.infer(normalized_audio)` --> **Here is memory leak was happend** 

and here is infer process code 
`def infer(self, audio):`
    `input_data = {self.input_tensor_name: audio}`
    `with torch.no_grad():`
  `    logits = self.infer_request.infer(input_data)[self.output_tensor]    `
    `return logits`

ive tried using with torch.no_grad() and gc.collect() and there is no any effect for the memory usage in this process.

Thank you for your helping

##### Issue submission checklist

 - [ ] I report the issue, it's not a question
   
 - [ ] I checked the problem with documentation, FAQ, open issues, Stack Overflow, etc and have not found solution
   
 - [ ] There is reproducer code and related data files: images, videos, models, etc.

**Comment by brmarkus** [comment]
Can you provide details about your environment, HW and SW, please?
To allow for reproduction and deeper analysis could you provide as much details, code, model, instructions of where to get the model and its conversion, please?
Can you use the model with an existing demo or sample from OpenVINO and Open-Model-Zoo and see the same increase in memory?

**Comment by ridhoalattas** [comment]
**Can you provide details about your environment, HW and SW, please?**
this is my enviroment :
torch==1.12.0
transformers==4.20.1 
soundfile==0.10.3.post1
pyctcdecode==0.1.0 
jiwer==2.3.0
pydub==0.25.1
SpeechRecognition==3.8.1
openvino==2022.1


**To allow for reproduction and deeper analysis could you provide as much details, code, model, instructions of where to get the model and its conversion, please?**
im convert my model with model_converter from demos, and got FP32 and FP16 and im trying using those two model and got same problem that memory are leaky when inference process.
Here is my code 

## Load Model

core = Core()
model = inf.Wav2Vec(core, TRANSCRIPT_MODEL, DEVICE_TYPE, VOCAB)

## Preprocess audio
audio = AudioSegment.from_file(path)
audio_new = preprocess_pydub_audio(audio)
audio_bytes = audio_new.raw_data
audio_bytes = byte_to_wavbyte(audio_bytes)
audio_input, sample_rate = sf.read(BytesIO(audio_bytes))

input_values = processor(audio_input, sampling_rate=sample_rate, return_tensors="pt").input_values

## Inference with model (Here is the problem)
with torch.no_grad():
    character_probs = model.infer(input_values)

**Can you use the model with an existing demo or sample from OpenVINO and Open-Model-Zoo and see the same increase in memory?**
I do that things too just like demo ways, but sadly the memory increase are exacly same with my process

**Comment by ridhoalattas** [comment]
is there any suggestion for this case?

**Comment by Iffa-Meah** [comment]
@ridhoalattas it would be great if you could share your model files for us to asses on our side.
There's possibility that the leak is something specific to your code instead of OpenVINO.
Your model files would help us to verify that.

Using the [correct precision for the correct hardware](https://www.intel.com/content/www/us/en/developer/articles/technical/should-i-choose-fp16-or-fp32-for-my-deep-learning-model.html) might somehow affect the inferencing result too.

If you are using Dynamic shapes increasing memory over time is expected which you may refer [here](https://docs.openvino.ai/latest/openvino_docs_OV_UG_DynamicShapes.html).

**Comment by Iffa-Meah** [comment]
Closing issue, feel free to re-open or start a new issue if additional assistance is needed.
